### DISCUSSION TITLE
Is there a way to start the web ui and let it listen to 0.0.0.0?

### DISCUSSION THREAD

**Discussion by jiayhuang** [open-webui/open-webui#3255] [General]
Thanks for developing this tool, this is awesome! I have a very small feature requests:

**Is your feature request related to a problem? Please describe.**
I am trying to serve the web UI on my local server and let my other device access it. What I would expect is to have some flag like HOST to let the web UI listen to 0.0.0.0. Then my other device could access it. 

**Describe the solution you'd like**
I tried to set HOST, but it seems to set the backend listening HOST instead of the FE node listening HOST as far as I can tell. And even setting the HOST as 0.0.0.0, I can still only access it from localhost:3000. Not sure if there is already some other recommended way I didn't find can help serving it and let me access it from other devices or I can try to add a FE_HOST flag.  :D

**Comment by TheTerrasque** [comment]
Have a look at bottom of https://github.com/open-webui/open-webui/blob/main/backend/start.sh

**Reply by alpha754293** [reply]
> Have a look at bottom of https://github.com/open-webui/open-webui/blob/main/backend/start.sh

I just tried starting the open-webui using this command:
`docker run -d -p 3000:8080 --gpus all --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:cuda`

I am able to access the open-webui via http://localhost:3000, but if I try to access it via the system's local IP:3000, that doesn't work.

Pardon my stupid question, but how would I modify the open-webui start command so that I would be able to access the open-webui docker container from other systems on my local network?

I tried reading the Docker documentation and since I am neither a programmer nor a developer, I couldn't really make heads nor tails to be able to combine the two (the Docker documentation and the open-webui start command).

Your help is greatly appreciated.

Thank you.

*edit*
I even tried to use Codestral:22b model to help me modify the `docker run` command such that it produced this:
`docker run -d --net=host --gpus all --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:cuda`

To try and get it so that the open-webui container will be able to listen on 0.0.0.0 to no avail.

It looks like that this command took the open-webui from http://localhost:3000 to http://localhost:8080 instead as Docker said that with the `--net=host` option, it removed the port specification in the previous `docker run` command. (I didn't know that.)

**Reply by envis10n** [reply]
Tiny bit of a necro, but what the prior post was pointing to is that it uses the $HOST environment variable to determine the binding address. If this variable is not present it should default to 0.0.0.0 (all interfaces). Go back to the original docker command, and add `-e HOST='0.0.0.0'` and see if it works.

This should set the HOST env inside the container, which is what you want.

**Comment by nickeisenberg** [comment]
Did you ever figure out how to do this?
